src/shogun/lib/Matrix.h

```diff
@@ -0,0 +1,57 @@
+#ifndef MATRIX_H__
+#define MATRIX_H__
+
+#include <shogun/lib/common.h>
+#include <shogun/lib/DataType.h>
+#include <shogun/lib/SGMatrix.h>
+
+namespace shogun
+{
+
+class Matrix {
+
+	void *m_data;
+	EPrimitiveType m_ptype;
+	index_t m_rows, m_cols;
```

Comment:
I like the minimal interface.
There should be in and out wrappers for SG* of course (later)